### ISSUE TITLE
[BUG] treemacs-follow-mode stopped working

### ISSUE THREAD

**Issue by jsravn** [hlissner/doom-emacs#1767]
**Describe the issue**

Updated to latest doom today, and `treemacs-follow-mode` has stopped working with the below error.

<details><pre>
Debugger entered--Lisp error: (wrong-type-argument stringp nil)
  expand-file-name(nil)
  vc-file-getprop(nil git-root)
  vc-git-root(nil)
  apply(vc-git-root nil)
  vc-call-backend(Git root nil)
  treemacs--git-status-process-function(nil)
  treemacs--git-status-process(nil #s(treemacs-project :name "console-docs" :path "/home/james/lightbend/console-docs" :path-status local-readable))
  treemacs-find-file-node("/home/james/lightbend/console-docs/src/main/paradox/installation/es.md" #s(treemacs-project :name "console-docs" :path "/home/james/lightbend/console-docs" :path-status local-readable))
  treemacs--follow()
  treemacs--init()
  (progn (fset (quote treemacs-workspace->is-empty\?) vnew) (treemacs--init))
  (unwind-protect (progn (fset (quote treemacs-workspace->is-empty\?) vnew) (treemacs--init)) (fset (quote treemacs-workspace->is-empty\?) old))
  (let* ((vnew (symbol-function (quote ignore))) (old (symbol-function (quote treemacs-workspace->is-empty\?)))) (unwind-protect (progn (fset (quote treemacs-workspace->is-empty\?) vnew) (treemacs--init)) (fset (quote treemacs-workspace->is-empty\?) old)))
  (let ((origin-buffer (current-buffer))) (let* ((vnew (symbol-function (quote ignore))) (old (symbol-function (quote treemacs-workspace->is-empty\?)))) (unwind-protect (progn (fset (quote treemacs-workspace->is-empty\?) vnew) (treemacs--init)) (fset (quote treemacs-workspace->is-empty\?) old))) (let ((--dolist-tail-- (treemacs-workspace->projects (treemacs-current-workspace)))) (while --dolist-tail-- (let ((project (car --dolist-tail--))) (treemacs-do-remove-project-from-workspace project) (setq --dolist-tail-- (cdr --dolist-tail--))))) (save-current-buffer (set-buffer origin-buffer) (let ((project-root (or (doom-project-root) default-directory))) (treemacs-do-add-project-to-workspace (treemacs--canonical-path project-root) (doom-project-name project-root))) (setq treemacs--ready-to-follow t) (if (or treemacs-follow-after-init treemacs-follow-mode) (progn (treemacs--follow)))))
  +treemacs--init()
  (if (eq val (quote visible)) (progn (delete-window (treemacs-get-local-window))) (+treemacs--init))
  (let* ((val (treemacs-current-visibility))) (if (eq val (quote visible)) (progn (delete-window (treemacs-get-local-window))) (+treemacs--init)))
  +treemacs/toggle()
  funcall-interactively(+treemacs/toggle)
  call-interactively(+treemacs/toggle nil nil)
  command-execute(+treemacs/toggle)
</pre></details>

**Steps to reproduce**
1. Enable `(treemacs-follow-mode 1)`
2. Navigate to projectile project
3. Open treemacs (SPC-o-p).

**System information**
<details><pre>
emacs   version    26.3
        features   XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS GLIB NOTIFY ACL GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS LIBSYSTEMD LCMS2
        build      Aug 29, 2019
        buildopts  (--prefix=/usr --sysconfdir=/etc --libexecdir=/usr/lib --localstatedir=/var --with-x-toolkit=gtk3 --with-xft --with-modules 'CFLAGS=-march=x86-64 -mtune=generic -O2 -pipe -fno-plt' CPPFLAGS=-D_FORTIFY_SOURCE=2 LDFLAGS=-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now)
doom    version    2.0.9
        build      HEAD -> develop, origin/develop, origin/HEAD 23509d22 2019-09-08 21:00:33 -0400
system  type       gnu/linux
        config     x86_64-pc-linux-gnu
        shell      /usr/bin/zsh
        uname      Linux 5.2.11-arch1-1-ARCH #1 SMP PREEMPT Thu Aug 29 08:09:36 UTC 2019 x86_64
        path       (~/.nvm/versions/node/v10.15.3/bin ~/.pyenv/shims ~/.jabba/jdk/adopt@1.11.0-3/bin ~/.pyenv/bin ~/.cargo/bin ~/.cabal/bin ~/.local/bin ~/go/bin ~/godev/bin ~/.emacs.d/bin /opt/google-cloud-sdk/bin /usr/local/bin /usr/local/sbin /usr/bin /usr/lib/jvm/default/bin /usr/bin/site_perl /usr/bin/vendor_perl /usr/bin/core_perl /usr/lib/emacs/26.3/x86_64-pc-linux-gnu)
config  envfile    nil
        elc-files  0
        modules    (:completion company helm :ui doom doom-dashboard doom-quit fill-column hl-todo modeline nav-flash ophints (popup +all +defaults) pretty-code treemacs unicode vc-gutter vi-tilde-fringe window-select :editor evil file-templates fold multiple-cursors rotate-text snippets :emacs dired electric vc :term term :tools docker editorconfig eval flycheck (lookup +docsets) lsp magit make :lang data emacs-lisp (go +lsp) javascript markdown (org +dragndrop +ipython +pandoc +present) (python +lsp +pyenv) (scala +lsp) sh :config default)
        packages   (n/a)
        elpa-packages (n/a)
</pre></details>

**Comment by hlissner** [comment]
I cannot reproduce this.

1. Can you reproduce this for *any* projectile project?
2. Can you reproduce this in vanilla Emacs and vanilla Doom (see [how to use Doom's sandbox](https://gist.github.com/hlissner/3910fcf5ef14ab77657f269ba9b37f94))?
3. Does the error persist if you disable the `:ui doom` module? (Remember to run `doom refresh` and restart Emacs afterwards)

**Comment by jsravn** [comment]
Dug a bit more:

1. It only seems to happen on one of my projects, afaict. I'm not sure what triggers it exactly. But I can open a specific file (a markdown file) and toggle treemacs to reproduce it. When it triggers, follow mode is broken for that project until I restart emacs or open another project.
2. I cannot reproduce it on vanilla Doom - follow mode works fine in the project. However on C-c C-p, I can reproduce it (same error as I listed above). Open project, open markdown file, toggle treemacs close/open and get error.
3. Yes it persists after disabling `:ui doom`.

**Comment by jsravn** [comment]
I'm fairly certain this is a treemacs bug not a doom bug. So feel free to close.

**Comment by Alexander-Miller** [comment]
What's the project look like, can you post a screenshot?

**Comment by JakDar** [comment]
I'm having the same errors. 
It occurs only when treemacs-collapse-dirs is > 0 and your current file is under collapsed dir.

**Comment by jsravn** [comment]
That seems to confirm what I'm seeing - if I open a file at the project root it works fine, if I open one farther down it fails.

**Comment by jsravn** [comment]
![image](https://user-images.githubusercontent.com/522123/64613617-7fa8bf80-d3ce-11e9-8324-87dc6089eccd.png)

README.md is fine, but if I open anything farther down and then toggle treemacs close/open I get the error.

**Comment by Alexander-Miller** [comment]
Yep, looks like some actions leave the internals in a bad state. I'll investigate.

**Comment by Alexander-Miller** [comment]
Pushed a fix. Try again with the latest version.

**Comment by jsravn** [comment]
@Alexander-Miller Thanks, it seems to be fixed in latest treemacs. Although it seems to take much longer now to open the project window - like ~3 seconds.
